subsys/bluetooth/mesh/light_ctrl_srv.c

```diff
@@ -572,10 +572,18 @@ static void reg_step(struct k_work *work)
 
 		srv->reg.prev = output;
 		atomic_set_bit(&srv->flags, FLAG_REGULATOR);
-		light_set(srv, light_to_repr(output, LINEAR), 0);
 	} else if (atomic_test_and_clear_bit(&srv->flags, FLAG_REGULATOR)) {
-		light_set(srv, light_to_repr(lvl, LINEAR), 0);
+		output = lvl;
+	} else {
+		return;
 	}
+
+	const struct bt_mesh_lightness_set set = {
```

Comment:
`transition` will have garbage uninitialized value due to local stack allocation. That causes unpredictable behavior within `bt_mesh_model_transition_time`

Reply:
Follow convention established elsewhere in mesh model stack that a NULL transition pointer represents immediate transition. It is handled correctly by bt_mesh_model_transition_time. May want to check that .h documents this convention correctly for backend callbacks, as well as any log statement.

Reply:
That's the truth but I have been spoken about a different things.
[ISO/IEC 9899:TC3 WG14/N1256](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf) (C99 standard) section 6.7.8 clause 10:
```
If an object that has automatic storage duration is not initialized explicitly, its value is
indeterminate.
```
To have the behavior that you mentioned you need to have:
```
const struct bt_mesh_lightness_set set = {
		.lvl = light_to_repr(output, LINEAR),
                .transition = NULL
	};
```

Reply:
Taken care of by 6.7.8 clause 21
If a structure variable is partially initialized, all the uninitialized structure members are implicitly initialized to zero no matter what the storage class of the structure variable is.